testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveCompatibility.java

```diff
@@ -184,8 +183,11 @@ public void testTimestampFieldWrittenByOptimizedParquetWriterCannotBeReadByHive(
             setSessionProperty(onTrino().getConnection(), "hive.timestamp_precision", hiveTimestampPrecision.name());
             onTrino().executeQuery("INSERT INTO " + tableName + " VALUES ('" + hiveTimestampPrecision.name() + "', TIMESTAMP '2021-01-05 12:01:00.111901001')");
             // Hive expects `INT96` (deprecated on Parquet) for timestamp values
-            assertQueryFailure(() -> onHive().executeQuery("SELECT a_timestamp FROM " + tableName + " WHERE timestamp_precision = '" + hiveTimestampPrecision.name() + "'"))
-                    .hasMessageMatching(".*java.lang.ClassCastException: org.apache.hadoop.io.LongWritable cannot be cast to org.apache.hadoop.hive.serde2.io.(TimestampWritable|TimestampWritableV2)");
+            int pow = (int) Math.pow(10, 9 - hiveTimestampPrecision.getPrecision());
+            int nanoOfSecond = IntMath.divide(111901001, pow, RoundingMode.HALF_UP) * pow;
```

Comment:
`111901001` magic code.
What is the meaning of this constant?

Reply:
That's the nanos part of the input timestamp and couple of lines above